gazebo_ros/include/gazebo_ros/node.hpp

```diff
@@ -177,6 +181,27 @@ Node::SharedPtr Node::CreateWithArgs(Args && ... args)
   // Add new node to the executor so its callbacks are called
   node->executor_->add_node(node);
 
+  // Parameter change callback
+  auto param_change_callback =
+    [node](std::vector<rclcpp::Parameter> parameters) {
+      auto result = rcl_interfaces::msg::SetParametersResult();
+      result.successful = true;
+
+      for (const auto & parameter : parameters) {
+        auto param_name = parameter.get_name();
+        if (param_name == "use_sim_time") {
+          RCLCPP_WARN(
+            node->get_logger(),
+            "use_sim_time will be ignored and messages will "
+            "continue to use simulation timestamps");
```

Comment:
Some rewording suggestions to make it clear this warning is with respect to a ROS parameter. Also, since we're just logging a warning, this statement is **not** necessarily true for third-party plugins.

```suggestion
            "use_sim_time parameter will be ignored by default plugins "
            "and ROS messages will continue to use simulation timestamps");
```

Alternatively, we could reject changes to `use_sim_time`, then any plugins using a ROS Clock associated with a `gazebo_ros::Node` will always use simulation time provided by the `/clock` topic. This might be a bit drastic though; I think warning is good improvement.

Reply:
Reworded the warning, and currently accepting ```use_sim_time``` changes with a warning.